src/components/reset-password.js

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+const users = [
+  {
+    email: "test@luxoft.com",
+    password: "luxoft"
+  }
+];
+export class ResetPassword extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      password: "",
+      cPassword: "",
+      message: "",
+      isSuccess: true
+    };
+    this.onChangeEmail = this.onChangeEmail.bind(this);
+    this.onChangePassword = this.onChangePassword.bind(this);
+    this.onChangecPassword = this.onChangecPassword.bind(this);
+    this.onClickSubmit = this.onClickSubmit.bind(this);
+    this.resetAlertMessage = this.resetAlertMessage.bind(this);
+    this.resetPassword = this.resetPassword.bind(this);
+  }
+
+  resetAlertMessage() {
+    setTimeout(() => {
+      this.setState({
+        message: ""
+      });
+    }, 1000);
+  }
+
+  resetPassword() {
+    this.setState({
+      password: "",
+      cPassword: ""
+    });
+  }
+
+  onChangeEmail(e) {
+    const { value: email } = e.target;
+    this.setState({
+      email
+    });
+  }
+
+  onChangePassword(e) {
+    const { value: password } = e.target;
+    this.setState({
+      password
+    });
+  }
+
+  onChangecPassword(e) {
+    const { value: cPassword } = e.target;
+    this.setState({
+      cPassword
+    });
+  }
+
+  onClickSubmit() {
+    const { email, password, cPassword } = this.state;
+    const userFound = users.findIndex(
+      user => user.email === email && user.password === password
+    );
+
+    if (userFound !== -1) {
+      this.props.history.push("/");
+    } else {
+      this.setState({
+        message: `Reset Failed, Please try with different credentials`,
+        isSuccess: false
+      });
+      this.resetAlertMessage();
+      this.resetPassword();
+    }
+  }
+
+  goToRegister = () => {
+    this.props.history.push("/register");
+  };
+
+  render() {
+    return (
+      <>
+        {this.state.message &&
+          (this.state.isSuccess ? (
```

Comment:
Is better to have this kind of validations outside the return.
We can create a variable or a function/method.